File: old_code/Source/OpSource/submit.c
/*----------------------------------------------------------------*
 * Program      : submit
 *
 * Language     : C
 *
 * Description  : Opens a connection to DD:INTOUT and writes
 *                JCL contained in char array.
 *
 * Called by    : cartsub
 *
 * Calls        : standard C functions
 *
 * Arguments    : (1) jcl     pointer to char array
 *              : (2) nlines  number of lines of JCL
 *              : (3) rc      error code = -3 if open failed
 *              :     jcllen  length of jcl, passed by Fortran
 *
 * Files        : //INTOUT DD SYSOUT=(A,INTRDR)
 *
 * $Workfile: submit.c$  $Folder: OpSource$
 * $Revision: 2$  $Date: 14/02/2012 09:13:56$
 *
 * Changes      :
 * $Log:
 *  2    Met_DB_Project 1.1         14/02/2012 09:13:56    Sheila Needham
 *       Replaces Fortran version for F95 compatability.
 *  1    Met_DB_Project 1.0         24/01/2012 11:36:26    Sheila Needham
 *       Initial version
 * $
 *
 *-----------------------------------------------------------------
 * (C) CROWN COPYRIGHT 2012 - MET OFFICE. All Rights Reserved.
 *
 * Met Office, United Kingdom.
 *
 * The use, duplication and disclosure of this code is strictly
 * prohibited without the permission of The Meteorological Database
 * Team at the above address.
 *-----------------------------------------------------------------*/
#include <stdio.h>
void submit_(char* jcl, int* lines, int *rc, int jcllen) {

  char* jclptr;
  char newline[]="\n";
  FILE *fp;

/* initialise jclptr to the start of the array (NB addresses) */

  jclptr=jcl;
  *rc = 0;

/* Open INTOUT file for writing          */

  if (( fp=fopen("DD:INTOUT","w,recfm=fb")) == NULL ) {
    perror("Open error on INTOUT in submit");
    *rc = -3;
    return;
  }
/* loop over lines of JCL writing to INOUT                 */

  for (int i=0; i<*lines ; i++) {
    fwrite(jclptr,jcllen,1,fp);
    fwrite(newline,1,1,fp);
    jclptr+=jcllen;
  }
  fclose(fp);
  return;
}

File: old_code/Source/OpSource/submit.f
      SUBROUTINE SUBMIT(CJCL,NL)                                                
                                                                                
!-----------------------------------------------------------------------        
!                                                                               
! ROUTINE       : SUBMIT
!                                                                               
! PURPOSE       : Submits a number of lines of jcl by writing                   
!                 directly to intout.                                           
!                                                                               
! CALLED BY     : CARTSUB
!                                                                               
! CALLS         : Nothing                                                       
!                                                                               
! I/O           : Output on unit 200 INTOUT                                     
!                                                                               
! ARGUMENTS     : (1) CJCL  : char*80 array of JCL lines to submit              
!               : (2) NL    : integer number of lines to submit                 
!
! REVISION INFO :
!
! $Revision: 1$
! $Date: 30/01/2006 20:24:42$
! $Source: /home/us0400/mdb/op/lib/source/RCS/submit.F,v $
!                                                                               
! CHANGE RECORD :                                                               
!                                                                               
! $Log:
!  1    Met_DB_Project 1.0         30/01/2006 20:24:42    Sheila Needham  
! $
! Revision 2.0  2001/07/03 10:44:01  usmdb
! Added copyright and modified header - S.Cox
!
! Revision 1.2  97/07/31  11:36:21  11:36:21  uspm (Pat McCormack)
! First revision for MVS
! 
! Revision 1.1  1997/07/04 13:39:14  uspm
! Initial revision
!
! 13-05-97  !A  : Add CLOSE(200) for MDB offline retrieval - S.Cox              
!
!-----------------------------------------------------------------------
! (C) CROWN COPYRIGHT 2001 - MET OFFICE. All Rights Reserved.
!
! Met Office, United Kingdom.
!
! The use, duplication and disclosure of this code is strictly
! prohibited without the permission of The Meteorological Database
! Team at the above address.
!-----------------------------------------------------------------------

      CHARACTER*80  CJCL(NL)                                                     
      CHARACTER*132 HEAD
                                                                                
      INTEGER       NL                                                           
                                                                                
      HEAD='
     &$Source: /home/us0400/mdb/op/lib/source/RCS/submit.F,v $
     &'//'$ $Date: 30/01/2006 20:24:42$ $Revision: 1$'
                                                                                
!-----------------------------------------------------------------------        
! connect unit 200 to DDNAME INTOUT                                             
!-----------------------------------------------------------------------        
                                                                                
      OPEN(200,FILE='INTOUT')                                                   
                                                                                
      WRITE(200,'(A80)')(CJCL(K),K=1,NL)                                        
                                                                                
      CLOSE(200)                                                      !A        
                                                                                
      RETURN                                                                    
      END
